dependencies.yaml

```
@@ -355,8 +355,10 @@ dependencies:
         packages:
           - numpy>=1.23,<3.0a0
           - zarr
-          # See https://github.com/zarr-developers/numcodecs/pull/475
-          - numcodecs !=0.12.0
+          # Pinned to the 0.13 for the following reasons:
+          #   https://github.com/zarr-developers/numcodecs/pull/475
+          #   0.14.0 doesn't offer python 3.10 support
+          - numcodecs>=0.13.0,<0.14.0a0
```

Not every version in this range needs to support every Python version that we build for. We only need _one_ compatible version of `numcodecs` in this range for each Python version we support.

Can we relax this?
```suggestion
          - numcodecs>=0.13.0
```